### ISSUE TITLE
>= Pandoc 3.2.1: warn about LaTeX writer changes

### ISSUE THREAD

**Issue by cscheid** [quarto-dev/quarto-cli#10850]
Pandoc 3.4 introduces a larger-than-usual change in the LaTeX writer that we should warn our users about. See [discussion](https://github.com/quarto-dev/quarto-cli/pull/10840#discussion_r1767647364).

**Comment by cwickham** [comment]
Do we have a plan for when and how we'll do this? 

It feels like extension authors might get complaints from their users as soon as we roll over to the 1.6 download. It would be nice to give them some advance warning.

**Comment by cscheid** [comment]
True. I think the best we can do is:

- point users to the actual Pandoc commit that made the change: https://github.com/jgm/pandoc/commit/26b25a4428815b04c255e33e95ee86ca7b6ee30e

- explain to them that they if they override the `graphics.tex` partial, they need to update (it's `src/resources/formats/pdf/pandoc/graphics.tex` in our repo)

**Comment by mcanouil** [comment]
If it helps.

- https://github.com/cmarquardt/quarto-simple-article
- https://github.com/gadenbuie/countdown/quarto
- https://github.com/kelly-sovacool/asm-msystems
- https://github.com/smnnlt/sportrxiv


```sh
CSV_FILE="https://raw.githubusercontent.com/mcanouil/quarto-extensions/refs/heads/main/extensions/quarto-extensions.csv"
curl -s "${CSV_FILE}" | while IFS= read -r REPO; do
  RESULT=$(gh api -X GET "repos/${REPO}/git/trees/main?recursive=1" --jq '.tree[] | select(.path | endswith("graphics.tex"))' 2>/dev/null)
  if [ $? -eq 1 ]; then
    RESULT=$(gh api -X GET "repos/${REPO}/git/trees/master?recursive=1" --jq '.tree[] | select(.path | endswith("graphics.tex"))' 2>/dev/null)
  fi
  if [ -n "${RESULT}" ]; then
    echo "https://github.com/${REPO}"
  fi
done
```

**Comment by cwickham** [comment]
Using the old Pandoc syntax directly in a `template` would also cause problems, right?

https://github.com/mps9506/quarto-cv/blob/a52baf31f57a6b732e02c2520785da9b958b5821/_extensions/quarto-cv/quarto-cv.tex#L24-L28

@mcanouil Could you expand your search to include that kind of use?

**Comment by mcanouil** [comment]
Yes, but it needs a bit more work as this needs to search in the files.

**Comment by mcanouil** [comment]
Here is an alternative version to find keyword in files (tex):

- https://github.com/christopherkenny/nature
- https://github.com/cmarquardt/quarto-simple-article
- https://github.com/dfolio/quarto-ieee
- https://github.com/dialoa/first-line-indent
- https://github.com/ihrke/mdpi
- https://github.com/juliantao/quarto-asce
- https://github.com/mps9506/quarto-cv
- https://github.com/nareal/naletter
- https://github.com/nareal/nastatement
- https://github.com/nmfs-opensci/quarto_titlepages
- https://github.com/quarto-journals/biophysical-journal
- https://github.com/quarto-journals/elsevier
- https://github.com/quarto-journals/jasa
- https://github.com/quarto-journals/plos
- https://github.com/sebdunnett/quarto-iop
- https://github.com/smnnlt/sportrxiv

```sh
#!/usr/bin/env bash

CSV_FILE="https://raw.githubusercontent.com/mcanouil/quarto-extensions/refs/heads/main/extensions/quarto-extensions.csv"

includes_keyword() {
  local repo=$1
  local branch=$2
  local keyword=$3

  tex_files=$(gh api -X GET "repos/${repo}/git/trees/${branch}?recursive=1" --jq '.tree[] | select(.path | endswith(".tex")) | .path' 2>/dev/null)
  if [[ "${tex_files}" == *'"status":"404"'* ]]; then
    echo "${tex_files}"
    return 0
  fi
  while IFS= read -r file; do
    file_content=$(gh api -X GET "repos/${repo}/contents/${file}?ref=${branch}" --jq '.content' 2>/dev/null | base64 --decode)
    if [[ "${file_content}" == *"${keyword}"* ]]; then
      echo "${repo}"
      return 0
    fi
  done <<< "${tex_files}"

  return 1
}

curl -s "${CSV_FILE}" | while IFS= read -r REPO; do
  RESULT=$(includes_keyword "${REPO}" "main" "graphicx")
  if [[ "${RESULT}" == *'"status":"404"'* ]]; then
    RESULT=$(includes_keyword "${REPO}" "master" "graphicx")
  fi
  if [[ "${RESULT}" == *'"status":"404"'* ]]; then
    echo "${REPO} has not main or master branch."
  fi

  if [[ -n "${RESULT}" && "${RESULT}" != *'"status":"404"'* ]]; then
    echo "https://github.com/${RESULT}"
  fi
done
```

**Comment by mcanouil** [comment]
Changing the keyword to look for:

```sh
KEYWORD="\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}"
````

- https://github.com/cmarquardt/quarto-simple-article
- https://github.com/dfolio/quarto-ieee
- https://github.com/dialoa/first-line-indent
- https://github.com/juliantao/quarto-asce
- https://github.com/mps9506/quarto-cv
- https://github.com/nareal/naletter
- https://github.com/nareal/nastatement
- https://github.com/nmfs-opensci/quarto_titlepages
- https://github.com/smnnlt/sportrxiv

**Comment by cderv** [comment]
We're working on 1.7 so I think this can be claused now. We'll do better for future change like that.
